users/models.py

```diff
@@ -1,3 +1,385 @@
+        """
+        Retrieve a queryset filtering only active users.
+
+        Returns:
+            QuerySet: A queryset of users with is_active=True.
+        """
+        return super().get_queryset().filter(is_active=True)
+
+
+class CustomUserManager(BaseUserManager):
+    """Manager for creating regular and super users."""
+
+    def create_user(self, email, password=None, **other_fields):
+        """
+        Create and save a regular user with the given email and password.
+
+        Args:
+            email (str): The user's email address.
+            password (str, optional): The user's password.
+            **other_fields: Additional fields for the user model.
+
+        Raises:
+            ValueError: If email is not provided or invalid.
+
+        Returns:
+            User: The created user instance.
+        """
+        if not email:
+            raise ValueError("Email required.")
+        validate_email(email)
+        email = self.normalize_email(email)
+        user = self.model(email=email, **other_fields)
+        user.set_password(password)
+        user.save(using=self._db)
+        return user
+
+    def create_superuser(self, email, password=None, **other_fields):
+        """
+        Create and save a superuser with the given email and password.
+
+        Args:
+            email (str): The superuser's email address.
+            password (str, optional): The superuser's password.
+            **other_fields: Additional fields for the user model.
+
+        Returns:
+            User: The created superuser instance.
+        """
+        other_fields.setdefault('is_staff', True)
+        other_fields.setdefault('is_superuser', True)
+        return self.create_user(email, password, **other_fields)
+
+
+class User(AbstractBaseUser, PermissionsMixin):
+    """
+    Custom user model.
+
+    Attributes:
+        user_id (AutoField): Primary key.
+        first_name (str): User's first name.
+        last_name (str): User's last name.
+        email (str): User's email address (unique).
+        password (str): Hashed password.
+        user_phone (str, optional): User's phone number.
+        title (str, optional): User's job title.
+        role (str): User role.
+        created_at (datetime): Record creation timestamp.
+        updated_at (datetime): Record last update timestamp.
+        is_active (bool): User active status.
+        is_staff (bool): User staff status.
+    """
+    
+    groups = models.ManyToManyField(
+        Group,
+        related_name='custom_users', 
+        blank=True,
+        help_text='The groups this user belongs to.',
+        verbose_name='groups',
+        related_query_name='custom_user',
+    )
+    user_permissions = models.ManyToManyField(
+        Permission,
+        related_name='custom_users',
+        blank=True,
+        help_text='Specific permissions for this user.',
+        verbose_name='user permissions',
+        related_query_name='custom_user',
+    )
```

Comment:
Is there a specific reason for redefining the groups and user_permissions fields with custom related_names? If not, you might stick with Django's defaults for better compatibility with Django admin and permissions.